### PR TITLE
Update/libp2p v0.2.1

### PR DESCRIPTION
**Pull request by frrist**
## Wat
This PR updates go-libp2 (and its deps) to v0.2.1 

## Status
1 test failure
### Test Failures:
  - `TestMigrateSomeRepo/Happy_path:_valid_migration_passes_validation`
This test will be skipped until https://github.com/filecoin-project/go-filecoin/issues/3133 is addressed.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -414,7 +414,8 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 	}
 
 	// Set up libp2p network
-	fsub, err := libp2pps.NewFloodSub(ctx, peerHost)
+	// TODO: we probably want to signe messages
+	fsub, err := libp2pps.NewFloodSub(ctx, peerHost, libp2pps.WithMessageSigning(false))
```
Comment (frrist) [inline node/node.go]: Issue for this here: https://github.com/filecoin-project/go-filecoin/issues/3124 will add to comment in code as well

**Comment by hannahhoward** [comment]
LGTM.... though of course this is hard to truly know if we caught everything...

File: tools/migration/migrations/repo-1-2/migration_test.go [hunk 2]
```diff
@@ -67,6 +67,12 @@ func TestMigrateSomeRepo(t *testing.T) {
 	require.NoError(t, os.Symlink(repoDir, repoSymLink))
 
 	t.Run("Happy path: valid migration passes validation", func(t *testing.T) {
+		// TODO the upgrade from go-cidv0.0.1 to go-cidv0.0.3 has broken this test
+		// as the encoding has changed from base58 to base32. This causes key
+		// lookups in the chain store to fail since the CID's are no longer
+		// the expected values.
+		// This skip can be removed when go-filecoin#3133 is addressed.
+		t.SkipNow()
```
Comment (frrist) [inline tools/migration/migrations/repo-1-2/migration_test.go]: To the reviewer: does this sound reasonable to you?

Comment (anorth) [reply tools/migration/migrations/repo-1-2/migration_test.go]: Sure, we'll delete this migration and test eventually. It's functioning primarily as a demonstration.

**Pull request merged**